### ISSUE TITLE
Default values for Interval and ScrapeTimeout in ServiceMonitor are not working

### ISSUE THREAD

**Issue by kosmoz** [mmontes11/mariadb-operator#38]
Creating a `MariaDB` resource with the minimal `metrics` configuration allowed by the CRD, causes an error.

Metrics configuration:
```yaml
  metrics:
    exporter:
      image:
        repository: prom/mysqld-exporter
        tag: v0.14.0
    serviceMonitor:
      prometheusRelease: kube-prometheus-stack
```

Error:
```json
{
    "level": "error",
    "ts": 1675436926.6985223,
    "msg": "Reconciler error",
    "controller": "mariadb",
    "controllerGroup": "mariadb.mmontes.io",
    "controllerKind": "MariaDB",
    "mariaDB": {
        "name": "mariadb",
        "namespace": "mariadb-test"
    },
    "namespace": "mariadb-test",
    "name": "mariadb",
    "reconcileID": "b27c0a29-a642-42ab-aa4c-90778e24f3bd",
    "error": "error creating ServiceMonitor: 1 error occurred:\n\t* error creating Service Monitor: ServiceMonitor.monitoring.coreos.com \"mariadb\" is invalid: [spec.endpoints[0].scrapeTimeout: Invalid value: \"'10s'\": spec.endpoints[0].scrapeTimeout in body should match '^(0|(([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?)$', spec.endpoints[0].interval: Invalid value: \"'10s'\": spec.endpoints[0].interval in body should match '^(0|(([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?)$']\n\n",
    "stacktrace": "sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:234"
}
```

I think the problem has to do with quoting, as the resource looks like this after the failed reconciliation:

```yaml
  metrics:
    exporter:
      image:
        pullPolicy: IfNotPresent
        repository: prom/mysqld-exporter
        tag: v0.14.0
    serviceMonitor:
      interval: '''10s'''
      prometheusRelease: kube-prometheus-stack
      scrapeTimeout: '''10s''' 

```

The version used to reproduce this error was v0.0.6.

**Comment by mmontes11** [comment]
Hey @kosmoz, thanks for reporting this, good catch 🙏🏻 

This was indeed a problem caused by the kubebuilder defaults. It has been fixed here:
- https://github.com/mmontes11/mariadb-operator/pull/39

It will be shipped in the next `v0.0.7` release, probably next week.

**Comment by mmontes11** [comment]
@kosmoz actually upgrading to v0.6.5 helm chart should fix your issue:
https://github.com/mmontes11/mariadb-operator/releases/tag/helm-chart-0.6.5
